### PR TITLE
CAMEL-21127 - Migrate from Deprecated o.a.maven.plugins.annotations.Component

### PR DESCRIPTION
**Pull request by ammachado**
# Description

Migrate Camel plugins from `@org.apache.maven.plugins.annotations.Component`

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/projects/CAMEL/issues/CAMEL-21127) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.
- [x] I have run `mvn clean install -DskipTests` locally from root folder and I have committed all auto-generated changes.

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -3046,6 +3048,11 @@
                     <artifactId>maven-plugin-plugin</artifactId>
                     <version>${maven-plugin-plugin-version}</version>
                 </plugin>
+                <plugin>
+                    <groupId>org.eclipse.sisu</groupId>
+                    <artifactId>sisu-maven-plugin</artifactId>
+                    <version>${sisu-maven-plugin-version}</version>
+                </plugin>
```
Comment (zhfeng) [inline parent/pom.xml:3055]: Why it need to introduce `sisu-maven-plugin`?

Comment (ammachado) [reply parent/pom.xml:3055]: As per Maven's documentation: https://maven.apache.org/maven-jsr330.html#how-to-use-jsr-330-in-plugins

**Review by zhfeng — APPROVED** [review]

**Review by oscerd — APPROVED** [review]
This is fine, but I would postpone to 4.9.0

**Comment by davsclaus** [comment]
Yes lets do this after 4.8.0 release as we had troubles in the past with doing some last minute maven changes or other build system changes that caused an upcoming release to fail.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**